pkg/engine2/operational_eval/evaluator_test.go

```diff
@@ -0,0 +1,306 @@
+			initialState: []Vertex{
+				&propertyVertex{
+					Ref: construct.PropertyRef{
+						Resource: graphtest.ParseId(t, "a:a:a"),
+						Property: "prop2",
+					},
+				},
+			},
+			source: Key{Ref: construct.PropertyRef{
+				Resource: graphtest.ParseId(t, "a:a:a"),
+				Property: "prop1",
+			}},
+			target: Key{Ref: construct.PropertyRef{
+				Resource: graphtest.ParseId(t, "a:a:a"),
+				Property: "prop2",
+			}},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			testSol := enginetesting.NewTestSolution()
+			eval := NewEvaluator(testSol)
+			for _, v := range tt.initialState {
+				err := eval.graph.AddVertex(v)
+				assert.NoError(err)
+				err = eval.unevaluated.AddVertex(v)
+				assert.NoError(err)
+			}
+			err := eval.addEdge(tt.source, tt.target)
+			if tt.wantErr {
+				assert.Error(err)
+				return
+			}
+			assert.NoError(err)
+			_, err = eval.graph.Edge(tt.source, tt.target)
+			assert.NoError(err)
+			_, err = eval.unevaluated.Edge(tt.source, tt.target)
+			assert.NoError(err)
+		})
+	}
+}
+
+func TestEvaluator_enqueue(t *testing.T) {
+	tests := []struct {
+		name         string
+		changes      graphChanges
+		initialState []Vertex
+		want         map[Key][]Key
+		wantErr      bool
+	}{
+		{
+			name: "simple enqueue",
+			initialState: []Vertex{
+				&propertyVertex{
+					Ref: construct.PropertyRef{
+						Resource: graphtest.ParseId(t, "a:a:a"),
+						Property: "prop2",
+					},
+				},
+			},
+			changes: graphChanges{
+				nodes: map[Key]Vertex{
+					{Ref: construct.PropertyRef{
+						Resource: graphtest.ParseId(t, "a:a:a"),
+						Property: "prop1",
+					}}: &propertyVertex{
+						Ref: construct.PropertyRef{
+							Resource: graphtest.ParseId(t, "a:a:a"),
+							Property: "prop1",
+						},
+					},
+				},
+				edges: map[Key]set.Set[Key]{
+
+					{Ref: construct.PropertyRef{
+						Resource: graphtest.ParseId(t, "a:a:a"),
+						Property: "prop1",
+					}}: set.SetOf[Key](
+						Key{Ref: construct.PropertyRef{
+							Resource: graphtest.ParseId(t, "a:a:a"),
+							Property: "prop2",
+						}},
+					),
+				},
+			},
+			want: map[Key][]Key{
+				{Ref: construct.PropertyRef{
+					Resource: graphtest.ParseId(t, "a:a:a"),
+					Property: "prop1",
+				}}: []Key{
+					{Ref: construct.PropertyRef{
+						Resource: graphtest.ParseId(t, "a:a:a"),
+						Property: "prop2",
+					}},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			assert := assert.New(t)
+			testSol := enginetesting.NewTestSolution()
+			eval := NewEvaluator(testSol)
+			for _, v := range tt.initialState {
+				err := eval.graph.AddVertex(v)
+				assert.NoError(err)
+			}
+			err := eval.enqueue(tt.changes)
+			if tt.wantErr {
+				assert.Error(err)
+				return
+			}
+			assert.NoError(err)
+			adj, err := eval.graph.AdjacencyMap()
+			assert.NoError(err)
+			for k, v := range tt.want {
```

Comment:
Need to also check that there's nothing in the `AdjacencyMAp` that isn't in `tt.want` (Order check should be sufficient).